helmfile/charts/prometheus-alerts/templates/alerts/packets-dropped.yaml

```
@@ -0,0 +1,42 @@
+{{- if and .Values.defaultRules.create .Values.defaultRules.rules.networkpolicies }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ printf "%s-%s" (include "prometheus-alerts.fullname" .) "packets-dropped" | trunc 63 | trimSuffix "-" }}
+  labels:
+    app: {{ template "prometheus-alerts.name" . }}
+    {{- include "prometheus-alerts.labels" . | nindent 4 }}
+    {{- if .Values.defaultRules.alertLabels }}
+    {{- toYaml .Values.defaultRules.alertLabels | indent 4 }}
+    {{- end }}
+  {{- if .Values.defaultRules.annotations }}
+  annotations:
+    {{- toYaml .Values.defaultRules.annotations | indent 4 }}
+  {{- end }}
+spec:
+  groups:
+  - name: packets-dropped
+    rules:
+    - alert: PacketsDroppedToWorkload
+      annotations:
+        description: Number of packets dropped to a workload because no policies matched them
+        summary: Number of packets dropped to a workload
+      expr: |
+        rate(no_policy_drop_counter{type="tw", cluster=~".*", pod_namespace=~".*", exported_pod=~".*"}[1m]) > 0
+      for: 5m
```

I think we want to trigger instantly here (or at least less than the range used in the alert `1m`)

```suggestion
      for: 0m
```

Since otherwise we would not catch the situation where we just have a single package drop each 10 minutes.

---

Sure thing

---

As calico-account will report package drop for unavailable pods as well, i would like this alert to notify us when there are drop packages over a longer period of time. And not instantly.

---

Should we have a compromise with 1 policy matching for a longer period (lets say more than 20 drops over 3 hours with a `for` of 6h) and one that's shorter (more than 0 over 1m (with a `for` of 5m)
Then we would quickly catch big spikes and lower rates drops.

---

I like that idea :+1:

---

@Xartos @robinAwallace I've updated the alerts, can you check if that's what you meant ? Thanks

---

Perfection :ok_hand: